### DISCUSSION TITLE
Train Discriminator less than Generator

### DISCUSSION THREAD

**Discussion by paxhx** [Lightning-AI/pytorch-lightning#6526] [code help: CV] [answered]
I want to train my discriminator ones per 10 iterations but couldn't figure out how to implement it with lightning. Do you have any advice on this?

**Comment by zhhao1** [comment]
here is an example:
def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, closure,
˓→on_tpu=False, using_native_amp=False, using_lbfgs=False):
# update generator opt every 2 steps
if optimizer_i == 0:
optimizer.step(closure=closure, make_optimizer_step=(batch_nb % 2) == 0)
# update discriminator opt every 4 steps
if optimizer_i == 1:
optimizer.step(closure=closure, make_optimizer_step=(batch_nb % 4) == 0)

**Comment by akihironitta** [comment, marked as answer]
Check out the optimization docs. There are a few examples that may help you. https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#step-optimizers-at-arbitrary-intervals
